Documentation/raspi-spi.md

```diff
@@ -0,0 +1,135 @@
+
+## Basic Hardware SPI usage
+
+The most simplest code you can build to use SPI is the following:
+
+```csharp
+using System;
+using System.Device.Spi;
+
+namespace TestTest
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            SpiDevice spi = SpiDevice.Create(new SpiConnectionSettings(0));
+            spi.WriteByte(0x42);
+            var incoming = spi.ReadByte();
+        }
+    }
+}
+```
+
+This will open SPI0, with the default Chip Select. It will then write a byte to the MOSI pin and read 1 byte from the MISO pin. 
+
+If you get something like this, it means you need to check the next sections to activate your SPI0:
+
+```
+Unhandled exception. System.IO.IOException: Error 2. Can not open SPI device file '/dev/spidev0.0'.
+   at System.Device.Spi.UnixSpiDevice.Initialize()
+   at System.Device.Spi.UnixSpiDevice.WriteByte(Byte value)
+   at SpiTest.Program.Main(String[] args) in C:\tmp\TestTest\SpiTest\Program.cs:line 11
+Aborted
+```
+
+
+## Enabling SPI0 without Hardware Chip Select
+
+In very short, this is the line you'll need to add into the `/boot/config.txt` file:
+
+```bash
+sudo nano /boot/config.txt
+```
+
+Add the line: 
+
+```
+dtparam=spi=on
+```
+
+Save the file with `ctrl + x` then `Y` then `enter`
+
+Then reboot:
+
+```bash
+sudo reboot
+```
+
+This will enable SP0 where those are the pins which will be seleted, only Software Chip Select is activated with the default pins:
+
+| SPI Function | Header Pin | GPIO # | Pin Name |
+| --- | --- | --- | --- |
+| MOSI | 19 | GPIO10 | SPI0_MOSI |
+| MISO | 21 | GPIO09 | SPI0_MISO |
+| SCLK | 23 | GPIO11 | SPI0_SCLK |
+| CE0 | 24 | GPIO08 | SPI0_CE0_N |
+| CE1 | 26 | GPIO07 | SPI0_CE1_N |
+
+## Enabling any SPI with any Chip Select
+
+In order to activate  Chip Select, you'll need to add a specific dtoverlay on the `/boot/config.txt` file. If you've used the previous way of activating SPI0, you should comment the line `dtparam=spi=on` and us what follows using the `dtoverlay`configurations.
+
+Here is the table with the different options for SP0 and SP1 (please refer to the [Raspberry Pi docuementation](https://www.raspberrypi.org/documentation/hardware/raspberrypi/spi/) to activate other SPI)
+
+# SPI0
+
+The following dtoverlay definition can be [found here](https://github.com/raspberrypi/firmware/blob/7b99da75f55a5ad7d572ec4ebe4e8f9573deaee7/boot/overlays/README#L2437).
+
+| SPI # | Chip Select # | Header Pin | Default GPIO | Pin Name | 
+| --- | --- | --- | --- | --- | --- | --- |
+| SPI0 | CE0 | 24 | GPIO08 | SPI0_CE0_N |
+| SPI0 | CE1 | 26 | GPIO07 | SPI0_CE1_N |
+
+
+If you want to change the default pins for Chip Select 0 to the GPIO pin 27 (hardware 13), and let's say GPIO pin 22 (hardware 15) for Chip Select 1, just add this line:
```

Comment:
Are you sure you can use any pin for hardware CS? The "alternate pin mode" table suggests that these are fixed.

Reply:
The table is the default configuration. Advance configuration is what I describe. So seems it is not super clear. How would you rephrase then?

Reply:
See above - I think we're talking about three different ways here. I'll have to think about how to describe this in an understandable and clear way.